zebrad/tests/acceptance.rs

```diff
@@ -2606,3 +2606,83 @@ async fn state_format_test(
     }
     Ok(())
 }
+
+/// Test that fixed responses at zcashd for the `z_getsubtreesbyindex` method are the same as zebrad responses.
+///
+/// Require scyncronized zebrad cache state to query high heights.
+#[tokio::test]
+#[ignore]
+async fn rpc_z_getsubtreesbyindex_zcashd_test_vectors() -> Result<()> {
+    use serde_json::Value;
+
+    let _init_guard = zebra_test::init();
+
+    // We're only using cached Zebra state here, so this test type is the most similar
+    let test_type = TestType::UpdateZebraCachedStateWithRpc;
+    let network = Network::Mainnet;
+
+    let (mut zebrad, zebra_rpc_address) = if let Some(zebrad_and_address) =
+        spawn_zebrad_for_rpc(network, "fully_synced_rpc_test", test_type, true)?
+    {
+        tracing::info!("running fully synced zebrad RPC test");
+
+        zebrad_and_address
+    } else {
+        // Skip the test, we don't have the required cached state
+        return Ok(());
+    };
+
+    // Wait for zebrad to load the full cached blockchain, locally this can take up to 3 minutes.
+    std::thread::sleep(Duration::from_secs(180));
```

Comment:
Waiting for a fixed amount of time is unreliable, because if the network or CI machine is slow, the test will fail. Instead, wait until Zebra logs that it is ready.

The test can still fail if it is upgrading an older cached state, so we might need to wait for that log as well. (Or maybe it's ok if all the cached states have been upgraded already. We can wait and see if it passes CI.)

```suggestion
    zebrad.expect_stdout_line_matches(SYNC_FINISHED_REGEX)?;
```

Reply:
I think this is a good suggestion and i appreciate it. I will apply, thanks.